aptos-move/framework/aptos-stdlib/sources/optional_aggregator.move

```diff
@@ -71,6 +76,48 @@ module aptos_std::optional_aggregator {
         }
     }
 
+    /// Switches between parallelizable and non-parallelizable implementations.
+    public(friend) fun switch(optional_aggregator: OptionalAggregator): OptionalAggregator {
+        let value = read(&optional_aggregator);
+        let new_optional_aggregator = switch_and_zero_out(optional_aggregator);
+        add(&mut new_optional_aggregator, value);
+
+        new_optional_aggregator
+    }
+
+    /// Switches between parallelizable and non-parallelizable implementations, setting
+    /// the value of the new optional aggregator to zero.
+    fun switch_and_zero_out(optional_aggregator: OptionalAggregator): OptionalAggregator {
+        let OptionalAggregator { aggregator, integer } = optional_aggregator;
+
+        if (option::is_some(&aggregator)) {
+            // In this case we convert from Some(Agg), None to None, Some(Int).
+            // First, get the limit and destroy old aggregator/integer pair.
+            let limit = aggregator::limit(option::borrow(&aggregator));
```

Comment:
maybe we can have helper functions that destroy some(aggregator), none(integer), return limit and destroy none(aggregator), some(integer), return limit. Can be re-used in destroy function at well.

Alternatively, switching shouldn't happen very often, so if we don't care about hyper-efficiency, we could just re-use destroy function here. We could also have a separate function that gets the limit and call it before line 93. (the overhead is conditional in both limit and destroy).